src/lib/themes.js

```diff
@@ -61,6 +62,10 @@ export const themesEqual = (themeA, themeB) => {
     return true;
   }
 
+  if (Object.keys(themeA.colors).length !== Object.keys(themeB.colors).length) {
```

Comment:
As mentioned in the test, this is not the right check. You should be checking default colors + advanced colors, if any.